chia/full_node/mempool.py

```diff
@@ -403,15 +410,39 @@ def create_bundle_from_mempool_items(
                 continue
             try:
                 cost = uint64(0 if item.npc_result.conds is None else item.npc_result.conds.cost)
-                unique_coin_spends, cost_saving, unique_additions = eligible_coin_spends.get_deduplication_info(
-                    bundle_coin_spends=item.bundle_coin_spends, max_cost=cost
-                )
+                if skipped_items >= ELIGIBILITY_SKIP_THRESHOLD:
+                    # If we've encountered `ELIGIBILITY_SKIP_THRESHOLD`
+                    # transactions that don't fit in the remaining block size,
+                    # we want to keep looking for smaller transactions that
+                    # might fit, but we also want to avoid spending too much
+                    # time on potentially expensive ones, hence this shortcut.
+                    unique_coin_spends = []
+                    unique_additions = []
+                    for spend_data in item.bundle_coin_spends.values():
+                        if spend_data.eligible_for_dedup:
+                            raise Exception(f"Skipping transaction with eligible coin(s): {name.hex()}")
```

Comment:
just to make sure I understand this correctly; If we don't abort here, we would construct an invalid block, right? We would include a double-spend.

Also, we could just `continue` here, rather than throwing an exception, couldn't we?